core/services/ocr2/plugins/ocr2keeper/evm21/upkeepstate/store.go

```diff
@@ -119,8 +134,14 @@ func (u *upkeepStateStore) Start(pctx context.Context) error {
 					}
 
 					ticker.Reset(utils.WithJitter(u.cleanCadence))
+				case <-flushTicker.C:
+					u.flush(ctx)
+				case err := <-u.errCh:
+					u.lggr.Errorw("error inserting records", "err", err)
 				case <-ctx.Done():
-
+					u.flush(ctx)
+					u.doneCh <- struct{}{}
```

Comment:
nit. couldn't find anyone reading from this channel?

Reply:
Note that I'm making some changes that affects it in https://github.com/smartcontractkit/chainlink/pull/10535/files

Reply:
@infiloop2 currently the tests read from this to check that the store has stopped before proceeding